ovos_media_plugin_spotify/__init__.py

```diff
@@ -15,9 +16,86 @@ def __init__(self, config, bus=None):
         super().__init__(config, bus)
         self.spotify = SpotifyClient()
         self._paused = False
-        self.ts = 0
+        self._last_sync_ts = 0
+        self._track_len = 0
+        self._track_pos = 0
         self.device_name = self.config.get("identifier")  # device name in spotify
 
+        self.bus.on("spotifyd.start", self.on_spotify_start)
+        self.bus.on("spotifyd.play", self.on_spotify_play)
+        self.bus.on("spotifyd.pause", self.on_spotify_pause)
+        self.bus.on("spotifyd.stop", self.on_spotify_stop)
+        self.bus.on("spotifyd.load", self.on_spotify_load)
+        self.bus.on("spotifyd.end_of_track", self.on_spotify_end)
+        self.bus.on("spotifyd.change", self.on_spotify_change)
+        self.bus.on("spotifyd.preloading", self.on_spotify_preloading)
+
+    ##################
+    # spotifyd hooks
+    def on_spotify_start(self, message: Message):
+        self._last_sync_ts = time.time()
+        self._now_playing = "spotify:track:" + message.data["track_id"]
+        self._track_pos = message.data["position"]  # milliseconds
+        self.bus.emit(message.forward("ovos.common_play.media.state",
+                                      {"state": 2}))  # loading media
```

Comment:
nit(non-blocking): consider an enum for this and all other state references

Reply:
it is in ovos-utils, just didnt use it to work with utils 0.0.38 otherwise would need 0.1.0, and dont want to duplicate that here again